src/hashes/ternary/curl_p/batched/mod.rs

```diff
@@ -0,0 +1,219 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! A batched version of the `CurlP` hash.
+
+mod bct;
+mod bct_curlp;
+
+use crate::hashes::ternary::{curl_p::CurlP, HASH_LENGTH};
+
+use bct::{BcTrit, BcTritArr, BcTritBuf};
+use bct_curlp::BctCurlP;
+
+use bee_ternary::{
+    raw::{RawEncoding, RawEncodingBuf},
+    Btrit, T1B1Buf, TritBuf,
+};
+
+/// The number of inputs that can be processed in a single batch.
+pub const BATCH_SIZE: usize = 8 * std::mem::size_of::<usize>();
+const HIGH_BITS: usize = usize::max_value();
+const NUM_ROUNDS: usize = 81;
+
+/// A hasher that can process several inputs at the same time in batches.
+///
+/// This hasher works by interleaving the trits of the inputs in each batch and hashing this
+/// interleaved representation. It is also able to fall back to the regular CurlP algorithm if
+/// required.
+pub struct BatchHasher<B: RawEncodingBuf> {
```

Comment:
Maybe we can come up with a more descriptive name. Right now it sounds very general, but it can actually only be used with `CurlP` at the moment. I'd be in favor of naming it `CurlPBatched` (or similar) analogous to `CurlPUnrolled`.

Reply:
I can rename it but it's actually the internal `BctCurlP` that has a similar API to regular curl. The `BatchHasher` is just a proxy to make things easier, I don't really think there is a way to make it similar to regular Curl without hurting ease of use.

Reply:
That makes sense. Should the name reflect that it uses Curl internally? Something like `CurlPBatchHasher`?

Reply:
Done